doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/user-guides/launching-clusters/on-premises.md

```diff
@@ -0,0 +1,116 @@
+ray start --address='<ip address>:6379'
+```
+
+The command will print out the address of the Ray GCS server that was started (the local node IP address plus the port number you specified). If the connection fails, check your firewall settings and network configuration.
+
+### Start Worker Nodes
+Then on each of the other nodes, run the following command to connect to the head node you just created.
+
+```
+ray start --address=<head-node-address:port>
+
+```
+Make sure to replace `head-node-address:port` with the value printed by the command on the head node (it should look something like 123.45.67.89:6379).
+
+Note that if your compute nodes are on their own subnetwork with Network Address Translation, to connect from a regular machine outside that subnetwork, the command printed by the head node will not work. You need to find an address that will reach the head node from the second machine. If the head node has a domain address like compute04.berkeley.edu, you can simply use that in place of an IP address and rely on the DNS.
+
+### Configure Nodes
+
+The ray start command also allows you configure the node's resources. For example, if you wish to specify that a machine has 10 CPUs and 1 GPU, you can do this with the flags --num-cpus=10 and --num-gpus=1. 
+See the [Configuration page](../../ray-core/configure.html#configuring-ray) for more information.
+
+### Trouble Shooting
+
+If you see `Unable to connect to GCS at ...`, this means the head node is inaccessible at the given `--address` (because, for example, the head node is not actually running, a different version of Ray is running at the specified address, the specified address is wrong, or there are firewall settings preventing access).
+
+If the connection fails, to check whether each port can be reached from a node, you can use a tool such as nmap or nc.
+
+```
+$ nmap -sV --reason -p $PORT $HEAD_ADDRESS
+Nmap scan report for compute04.berkeley.edu (123.456.78.910)
+Host is up, received echo-reply ttl 60 (0.00087s latency).
+rDNS record for 123.456.78.910: compute04.berkeley.edu
+PORT     STATE SERVICE REASON         VERSION
+6379/tcp open  redis?  syn-ack
+Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
+$ nc -vv -z $HEAD_ADDRESS $PORT
+Connection to compute04.berkeley.edu 6379 port [tcp/*] succeeded!
+```
+
+If the node cannot access that port at that IP address, you might see
+
+```
+$ nmap -sV --reason -p $PORT $HEAD_ADDRESS
+Nmap scan report for compute04.berkeley.edu (123.456.78.910)
+Host is up (0.0011s latency).
+rDNS record for 123.456.78.910: compute04.berkeley.edu
+PORT     STATE  SERVICE REASON       VERSION
+6379/tcp closed redis   reset ttl 60
+Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
+$ nc -vv -z $HEAD_ADDRESS $PORT
+nc: connect to compute04.berkeley.edu port 6379 (tcp) failed: Connection refused
+```
+
+## Using Ray Cluster Launcher 
```

Comment:
Should we put this section first, since we recommend this method if possible?

Reply:
Hmm, I've definitely observed that it is more common to use the manual setup than cluster launcher for on-prem setups.

Also see that it is non-consistent. I'll leave it as is but the copy editor should decide whether this should be flipped or something else should be done.

Reply:
Consider adding a TODO in plaintext with a link to this message to ensure the copy editor sees it. I’ll make sure all TODOs are removed before replacing old docs.